File: 1480. Running Sum of 1d Array/TypeScript/1480. Running Sum of 1d Array.ts
function runningSum(nums: number[]): number[] {

    let runningSums: number[] = []
    let runningSum: number = 0

    nums.forEach(function (num){

        runningSum += num
        runningSums.push(runningSum)

    }); 
    return runningSums   
};

let nums = [1,2,3,4];

runningSum(nums);

File: 1480. Running Sum of 1d Array/Python/1480. Running Sum of 1d Array.py
from typing import List
from abc import ABC, abstractmethod

class Solution(ABC):
    @property
    @abstractmethod
    def runningSum(self, nums: List[int]) -> List[int]:
        pass
    
class SolutionOne(Solution):
    def runningSum(nums: List[int]) -> List[int]:
        
        runningSums: List[int] = []
        runningSum: int = 0
        for num in nums:
            runningSum += num
            runningSums.append(runningSum)

        return runningSums   
    
if "__main__":
    nums1: List[int] = [1,2,3,4]
    nums2: List[int] = [1,1,1,1,1]
    nums3: List[int] = [3,1,2,10,1]

    solution = SolutionOne
    solution.runningSum(nums1)